pkg/apis/backup/v1alpha/backup_validate.go

```diff
@@ -0,0 +1,81 @@
+//
+// DISCLAIMER
+//
+// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Adam Janikowski
+//
+
+package v1alpha
+
+import "fmt"
+
+func (a *ArangoBackup) Validate() error {
+	if err := a.Spec.Validate(); err != nil {
+		return err
+	}
+
+	if err := a.Status.Validate(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *ArangoBackupSpec) Validate() error {
+	if a.Deployment.Name == "" {
+		return fmt.Errorf("deployment name can not be empty")
```

Comment:
This change does not apply to linter, because of `Error strings should not be capitalized` error in IDE

Reply:
That lint is there because they are trying to prevent double capitalization when the error message is printed out.

I'm not sure when and where Validate will be called, but looking at other things here, it looks like a lowercase version will be required because most of th time it's being appended to something or preceived by something, like a log message or formatting.

However, in that case @ajanikow you have to fix line 61. :) Which starts capitalized. :) Now I realize that in that case, RepositoryURL is a field, but so is Deployment.

Which means you have to decide now, if you want to say in the error more explicity something like this:

```go
fmt.Errorf("Deployment.Name can not be empty")
```

Which might be maybe preferred as it would indicate that a field name is being referenced here.